scope/xgb.py

```diff
@@ -0,0 +1,206 @@
+import datetime
+import os
+import pathlib
+from .models import AbstractClassifier
+
+#!git clone https://github.com/DTrimarchi10/confusion_matrix.git
+import numpy as np
+from xgboost import XGBClassifier
+import xgboost as xgb
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score, confusion_matrix, matthews_corrcoef, mean_absolute_error
+from sklearn.metrics import f1_score, ConfusionMatrixDisplay
+#plot_confusion_matrix, is deprecated, so import ConfusionMatrixDisplay instead
+from sklearn.metrics import roc_curve, auc, precision_recall_curve, classification_report, average_precision_score
+import pandas as pd
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import os
+import sys
+import copy
+import json
+
+
+class XGB(AbstractClassifier):
+    """Baseline model with a statically-defined architecture"""
+
+    def setup(
+        self,
+        **kwargs,
+    )
+
+    #I need to create a call for the run's specification:
+    ## ie xgbrun = {'datasetname': 'd15',
+      #      'classifiername': 'yso',
+       #     'featuresetname': 'f'}
+
+#modelprefix = xgbrun['datasetname'] +'.'+ xgbrun['classifiername'] +'.'+ xgbrun['featuresetname']
+#modelname = modelprefix + '.model'
+#fpfile = modelprefix + '_FPs.csv'
+#fnfile = modelprefix + '_FNs.csv'
+#density = modelprefix + '_density.png'
+#cmpng = modelprefix + '_cm.png'
+#histpng = modelprefix + '_hist.png'
+#impvars = modelprefix + '_impvars.png'
+#fnrfpr = modelprefix + '_fnrfpr.png'
+#recallpng = modelprefix + '_recall.png'
+#rocpng = modelprefix + '_roc.png'
+#jsonname = modelprefix + '.json'
+#txtfile = modelprefix + '.txt'
+#f = open(txtfile, "w")
+#f.write("dataset: %s\nclassifier: %s\nfeatureset: %s\n" % (xgbrun['datasetname'], xgbrun['classifiername'], xgbrun['featuresetname']))
+# LEAVE THIS UNTIL THE END WHEN RUNNING, so all data goes into file f: f.close()
+
+
+#READ IN DATASET
+
+#Choose features/labels/ids, seed/test size/train-test-split
+
+#Define params necessary for a training::
+params = {
+    # Parameters that we are going to tun? These are hyperparameters, so we only tune them with cross-validation rounds...
+    'max_depth':6,
+    'min_child_weight': 1,
+    'eta':.1,
+    'subsample': 0.7,
+    'colsample_bytree': 0.7,
+    # Other parameters
+#    'objective':'reg:squarederror',   # changed from reg:linear
+    'objective': 'binary:logistic',
+  #  'gpu_id': 0,
+  #  'tree_method': 'gpu_hist',
+    'eval_metric': 'auc'
+}
```

Comment:
These params could be passed as individual arguments to `setup()`. For example, see the analogous code in nn.py:
```
def setup(
        self,
        dense_branch: bool = True,
        conv_branch: bool = True,
        loss: str = "binary_crossentropy",
        optimizer: str = "adam",
        callbacks: tuple = ("reduce_lr_on_plateau", "early_stopping"),
        tag: Optional[str] = None,
        logdir: str = "logs",
        **kwargs,
    ):
```
Then I'd suggest collecting the inputs in a dictionary called `params` and storing it in `self.meta`, e.g. `self.meta['params'] = params`. This will help users setup xgboost models with different parameters using this function.